### PR TITLE
Add initial installer doc.

### PR DESCRIPTION
**Pull request by patrickdillon**
cc @staebler

### CODE REVIEW

File: docs/installer/index.md [hunk 1]
```diff
@@ -1,3 +1,65 @@
 # Installer
 
-TODO
\ No newline at end of file
+The [OpenShift Installer repo][repo] contains the code for the `openshift-install` Go binary as well as templates and documentation
+for user-provisioned infrastructure. `openshift-install` provides a user interface for the installation process on multiple cloud platforms.
+Users provide install configuration through an [`install-config.yaml`][install-config] file, which can be created interactively through the
+`openshift-install create install-config command. The installer uses this configuration to create Kubernetes manifests and ignition configs
```
Comment (elmiko) [inline docs/installer/index.md:6]: missing backtick here
```suggestion
`openshift-install create install-config` command. The installer uses this configuration to create Kubernetes manifests and ignition configs
```

File: docs/installer/index.md [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+[overview]: https://github.com/openshift/installer/blob/master/docs/user/overview.md
+[arch]: https://docs.openshift.com/container-platform/4.8/architecture/architecture-installation.html
+[contributing]: https://github.com/openshift/installer/blob/master/CONTRIBUTING.md
+[types]: https://github.com/openshift/installer/commit/590dc0c62d432d4d3ea1ca40aa84ba6f17bee780#diff-5dc50083bb8d12ad7439e48e73af24cc866f5fb59322144805cba65a8abdefc7
+[explain]: https://github.com/openshift/installer/blob/master/docs/dev/explain.md
+[asset-generation]: https://github.com/openshift/installer/blob/master/docs/design/assetgeneration.md
+[asset-example]: https://github.com/openshift/installer/pull/5333/commits/0577ab9ff94b2688645186674ec0eec0c8b5c190
+[tf-integration]: https://github.com/openshift/installer/pull/5333/commits/e99ecc6bf902542b207cd76274ec6f861279ab0e
+[tf-configs]: https://github.com/openshift/installer/pull/5333/commits/75ca60ff2483379d29fe6b78064783ccbd4ee6db
+[destroy:] https://github.com/openshift/installer/commit/0ef6c1367a42978059665be13e82a08ea74a7f37#diff-80f8e210f9e17a175682591d811168a5779aed19b3dee41f280014f2c2040127
```
Comment (elmiko) [inline docs/installer/index.md:64]: minor syntax error
```suggestion
[destroy]: https://github.com/openshift/installer/commit/0ef6c1367a42978059665be13e82a08ea74a7f37#diff-80f8e210f9e17a175682591d811168a5779aed19b3dee41f280014f2c2040127
```

**Review by elmiko — COMMENTED** [review]
thanks @patrickdillon , in general i think the content here is good and i like the heavy linking back to the installer docs.

i have a couple syntax nits, and i think it would be nice to pull the example links out of the body text and give them a separate bullet list or something. similar to how we've broken out the PR links here, https://notes.elmiko.dev/ocp-infrastructure-provider-onboarding-guide/cloud-controller-manager/

**Comment by patrickdillon** [comment]
Comments have been addressed

**Comment by elmiko** [comment]
thanks @patrickdillon , lgtm. apologies for the delay

**Pull request merged**